hamilton/io/default_data_loaders.py

```diff
@@ -6,9 +6,55 @@
 import pickle
 from typing import Any, Collection, Dict, Tuple, Type, Union
 
+import yaml
+
 from hamilton.io.data_adapters import DataLoader, DataSaver
 from hamilton.io.utils import get_file_metadata
 
+PrimitiveType = Union[str, int, bool, dict, list]
+
+
+@dataclasses.dataclass
+class YAMLDataLoader(DataLoader):
+    path: Union[str, pathlib.Path]
+
+    @classmethod
+    def applicable_types(cls) -> Collection[Type]:
+        return [str, int, bool, dict, list]
```

Comment:
can replace with PrimititeType?

Reply:
Idk why I thought this would fail lol, thanks for the review.

Reply:
Actually good point -- something to validate with an example. I could be wrong here.

Reply:
actually I think I'm wrong here. What you had was correct.